core/src/main/scala/org/apache/spark/deploy/security/HadoopAccessManager.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.security
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+
+/**
+ * Methods in [[HadoopAccessManager]] return scheduler-specific information related to how Hadoop
+ * delegation tokens should be fetched.
+ */
+private[spark] trait HadoopAccessManager {
```

Comment:
Thanks.  I'll make sure to do that in the future.  We happened to already have most of this code written internally before upstreaming to Apache, which is why I submitted before going through a design review.

As for this trait, I introduced it in order to parameterize `HadoopFSCredentialProvider` to support different behavior on YARN than on Mesos.  If you look at `YARNHadoopAccessManager`, you'll see that it uses YARN specific configuration variables.

I had originally tried to introduce both a `YARNHadoopFSCredentialProvider` and a `DefaultHadoopFSCredentialProvider` in order to support different behaviors, but as vanzin pointed out here: https://github.com/apache/spark/pull/17665#discussion_r112298650, we can't service load different providers on YARN than on Mesos.